src/defrag.c

```diff
@@ -308,12 +310,12 @@ Defrag4Reassemble(ThreadVars *tv, DefragTracker *tracker, Packet *p)
                 goto error_remove_tracker;
 
             hlen = frag->hlen;
-            ip_hdr_offset = frag->ip_hdr_offset;
+            ip_hdr_offset = tracker->ip_hdr_offset;
 
             /* This is the start of the fragmentable portion of the
              * first packet.  All fragment offsets are relative to
              * this. */
-            fragmentable_offset = frag->ip_hdr_offset + frag->hlen;
+            fragmentable_offset = tracker->ip_hdr_offset + frag->hlen;
```

Comment:
Need to look closer, but could this be a possible behaviour change not covered by a unit test?  2 overlapping packets with offset 0.  Instead of looking at the packet for each fragment, this will look at what was last set in the tracker.